### PR TITLE
Initial check in based on aborsavich/gogomarshal

### PR DESCRIPTION
**Pull request by johanbrandhorst**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,7 @@
 
 This repo contains a JSON marshaler suitable for use with the gRPC-Gateway
 when using `gogo/protobuf` types.
+
+## License
+
+95% of the code is copied from the gRPC-Gateway project, so their license applies.
```
Comment (awalterschulze) [inline README.md:8]: Is this true?
Just checking.

**Review by awalterschulze — APPROVED** [review]

File: jsonpb.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
+// Package gateway contains marshaling code extracted from
+// github.com/grpc-ecosystem/grpc-gateway/runtime and altered
+// to depend on gogo versions of the proto and jsonpb packages.
+package gateway
+
+import (
+	"bytes"
+	"encoding/json"
```
Comment (awalterschulze) [inline jsonpb.go:8]: Probably in for a following pull request, but do we need the standard encoding/json ?

Comment (johanbrandhorst) [reply jsonpb.go:8]: As you imply, I will investigate the warning printouts separately, and they may have something to do with using stdlib json.

File: README.md (hunk 1 above)
Comment (johanbrandhorst) [reply README.md:8]: > Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.

Easiest just to copy the LICENSE.

Comment (johanbrandhorst) [reply README.md:8]: https://github.com/grpc-ecosystem/grpc-gateway/blob/master/LICENSE.txt

**Pull request merged**